File: box.c
#include <stdio.h>

int main(int argc, char** argv) {
  
  int x = 4;
	    
		printf("+");
		for (int i = 0; i < x; i++) {
		    puts("-+");
		    for (int j = 0; j < i; j++) {
		        printf("  ");
		    }
		    
		    puts("| |");
		    for (int j = 0; j < i; j++) {
		        printf("  ");
		    }
		    
		    printf("+-+");
		}
  
  return 0;
}

File: box.rb
x = 4


string = "+"

i = 0
while i < x
    string += "-+" + "\n"
    j = 0
    while j < i
        string += "  "
        j += 1
    end
    string += "| |" + "\n"
    j = 0
    while j < i
        string += "  "
        j += 1
    end
    string += "+-+"
    i += 1
end

puts string
